File: 2D array/linear search.cpp
#include <bits/stdc++.h>
using namespace std;

bool searching(int arr[][4], int target)
{
    int index = 0;
    for (int i = 0; i < 3; i++)
    {
        for (int j = 0; j < 4; j++)
        {
            if (arr[i][j] == target)
            {
                return true;
            }
        }
    }
    return false;
}

int main()
{
    int arr[3][4], target = 13;
    cout << "Enter elements in the array: " << endl;
    for (int i = 0; i < 3; i++)
    {
        for (int j = 0; j < 4; j++)
        {
            cin >> arr[i][j];
        }
    }
    int x = searching(arr, target);
    if (x == 1)
    {
        cout << target << " is found in the array" << endl;
    }
    else
    {
        cout << target << " is not found in the array" << endl;
    }
    return 0;
}

File: recursion/day2/linearSearch.java

// package patterns.recursion.day2;
import java.util.*;

public class linearSearch {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enetr number of elements in the array: ");
        int n = sc.nextInt();
        int arr[] = new int[n];
        System.out.println("Enter elements in the array: ");
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        System.out.println("Enter the elements to find: ");
        int key = sc.nextInt();
        int ans = findKey(arr, n, key);
        if (ans == -1) {
            System.out.println(key + " is not found");
        } else {
            System.out.println(key + " is found at index " + ans);
        }
    }

    public static int findKey(int arr[], int n, int key) {
        int ans = 0;
        for (int i = 0; i < n; i++) {
            if (arr[i] == key) {
                return i;
            }
        }
        return -1;
    }
}
